test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -6571,6 +6569,20 @@ TEST_F(AtenXlaTensorTest, TestAsStridedWithOffset) {
   });
 }
 
+TEST_F(AtenXlaTensorTest, TestAsStridedWithInplaceCopy) {
+  torch::Tensor grad = torch::ones({4}, torch::TensorOptions(torch::kFloat));
+  std::vector<int64_t> size = {4};
+  std::vector<int64_t> stride = {1};
+  torch::Tensor output = torch::zeros({4}, grad.options());
+  output.as_strided(size, stride).copy_(grad);
+  ForEachDevice([&](const torch::Device& device) {
+    torch::Tensor xla_grad = CopyToDevice(grad, device);
+    torch::Tensor xla_output = torch::zeros({4}, xla_grad.options());
```

Comment:
This hasn't been copied to device, so I think the line below is testing:
<tensor not on device>.as_strided(size, stride).copy_(<tensor on device>);

We might want to test either instead or in addition:
<tensor on device>.as_strided(size, stride).copy_(<tensor on device>);

Reply:
xla_grad.options() contains the device, so the xla_output is on XLA device.
Otherwise the copy below would ask a pytorch CPU tensor to copy data from an XLA one, and it'd fail.